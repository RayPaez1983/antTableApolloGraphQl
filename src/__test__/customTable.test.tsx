import React from 'react';
import CustomTable from 'src/components/commons/customTable/customTable';
import dataMock from 'src/utils/mockCompaniesData.json';
import ColumnsMocked from 'src/__test__/mockData/tableMock';
import renderer from 'react-test-renderer';

it('SnapShot of the Table renders correctly', () => {
  const tree = renderer
    .create(<CustomTable data={dataMock.COMPANIES as []} columns={ColumnsMocked as []} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
