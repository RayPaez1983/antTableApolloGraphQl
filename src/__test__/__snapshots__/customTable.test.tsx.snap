// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot of the Table renders correctly 1`] = `
<div
  className="ant-col ant-col-24 sc-hLBbgP lzzud css-dev-only-do-not-override-sk7ap8"
  moveToRight={false}
  style={Object {}}
>
  <div
    onPointerDown={[Function]}
    role="presentation"
  >
    <div
      className="ant-table-wrapper css-dev-only-do-not-override-sk7ap8"
    >
      <div
        className="ant-spin-nested-loading css-dev-only-do-not-override-sk7ap8"
      >
        <div
          className="ant-spin-container"
        >
          <ul
            className="ant-pagination ant-pagination-simple ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-sk7ap8"
            style={Object {}}
          >
            <li
              className="ant-pagination-total-text"
            />
            <li
              aria-disabled={true}
              className="ant-pagination-prev ant-pagination-disabled"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex={null}
              title="Previous Page"
            >
              <button
                className="ant-pagination-item-link"
                disabled={true}
                tabIndex={-1}
                type="button"
              >
                <span
                  aria-label="left"
                  className="anticon anticon-left"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="left"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              className="ant-pagination-simple-pager"
              title="0/60"
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                size="3"
                type="text"
                value={0}
              />
              <span
                className="ant-pagination-slash"
              >
                /
              </span>
              60
            </li>
            <li
              aria-disabled={false}
              className="ant-pagination-next"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex={null}
              title="Next Page"
            >
              <button
                className="ant-pagination-item-link"
                disabled={false}
                tabIndex={-1}
                type="button"
              >
                <span
                  aria-label="right"
                  className="anticon anticon-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                    />
                  </svg>
                </span>
              </button>
            </li>
          </ul>
          <div
            className="ant-table ant-table-fixed-header ant-table-scroll-horizontal"
            data-testid="userTest"
          >
            <div
              className="ant-table-container"
            >
              <div
                className="ant-table-header"
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <table
                  style={
                    Object {
                      "tableLayout": "fixed",
                      "visibility": null,
                    }
                  }
                >
                  <colgroup>
                    <col
                      className="ant-table-selection-col"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    />
                  </colgroup>
                  <thead
                    className="ant-table-thead"
                  >
                    <tr>
                      <th
                        className="ant-table-cell ant-table-selection-column"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        scope="col"
                        style={Object {}}
                      >
                        <div
                          className="ant-table-selection"
                        >
                          <label
                            className="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-sk7ap8"
                          >
                            <span
                              className="ant-checkbox css-dev-only-do-not-override-sk7ap8 ant-checkbox-disabled"
                              style={Object {}}
                            >
                              <input
                                aria-label="Select all"
                                checked={false}
                                className="ant-checkbox-input"
                                disabled={true}
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onKeyPress={[Function]}
                                onKeyUp={[Function]}
                                type="checkbox"
                              />
                              <span
                                className="ant-checkbox-inner"
                              />
                            </span>
                          </label>
                        </div>
                      </th>
                    </tr>
                  </thead>
                </table>
              </div>
              <div
                className="ant-table-body"
                onScroll={[Function]}
                style={
                  Object {
                    "maxHeight": "calc(100vh - 360px)",
                    "overflowX": "auto",
                    "overflowY": "scroll",
                  }
                }
              >
                <table
                  style={
                    Object {
                      "minWidth": "100%",
                      "tableLayout": "fixed",
                      "width": "100vw",
                    }
                  }
                >
                  <colgroup>
                    <col
                      className="ant-table-selection-col"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    />
                  </colgroup>
                  <tbody
                    className="ant-table-tbody"
                  >
                    <tr
                      aria-hidden="true"
                      className="ant-table-measure-row"
                      style={
                        Object {
                          "fontSize": 0,
                          "height": 0,
                        }
                      }
                    >
                      <td
                        style={
                          Object {
                            "border": 0,
                            "height": 0,
                            "padding": 0,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "height": 0,
                              "overflow": "hidden",
                            }
                          }
                        >
                          Â 
                        </div>
                      </td>
                    </tr>
                    <tr
                      className="ant-table-placeholder"
                      style={
                        Object {
                          "display": null,
                        }
                      }
                    >
                      <td
                        className="ant-table-cell"
                        colSpan={null}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        rowSpan={null}
                        style={Object {}}
                      >
                        <div
                          className="ant-table-expanded-row-fixed"
                          style={
                            Object {
                              "left": 0,
                              "overflow": "hidden",
                              "position": "sticky",
                              "width": 0,
                            }
                          }
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
